esy-package-config/Resolution.re

```diff
@@ -5,7 +5,10 @@ type t = {
   resolution,
 }
 and resolution =
-  | Version(Version.t)
+  | VersionOverride({
+      version: Version.t,
+      override: option(Json.t),
+    })
```

Comment:
Just a small nit, instead of having `override` as an option inside `VersionOverride` 
can you keep the `Version` constructor as is and add a new constructor `VersionOverride({
      version: Version.t,
      override: Json.t
    })` ?

Reply:
The problem is that they behave identically, from the point of view of the solver, originally I had done that but then I endup duplicating a lot of logic.

But when I changed the Version field it was clear where changes were needed to move this around.